src/plots/plots.js

```diff
@@ -527,6 +563,9 @@ plots.supplyDataDefaults = function(dataIn, dataOut, layout) {
         basePlotModules = layout._basePlotModules = [],
         cnt = 0;
 
+    // Make this idempotent by emptying the existing array:
+    dataOut.length = 0;
```

Comment:
Sweet. Good to know.

I don't see `Plots.supplyDataDefaults` getting called anywhere in the combined diff. Is this an artifact of your attempt to by-pass the full `Plots.supplyDefaults` ?